### ISSUE TITLE
Allow to inject mobify.js into page instead of script load

### ISSUE THREAD

**Issue by puzrin** [mobify/mobifyjs#213]
It would be nice to have alternate init, when mobify script is completely injected into page (via asset manager).

For example, we do hybrid app, where client's html is loaded only once, and then everything switches to ajax http://dev.nodeca.com/ . I this case, injecting customized build of mobify.js directly into page looks quite reasonable.

Could you provide alternate initializer for that?

**Comment by jansepar** [comment]
Hi @puzrin, just to be clear, what you want is a way to load mobify.js with the entire file inlined in the page, rather then loading mobify.js using an external file?

**Comment by puzrin** [comment]
Yes. Exactly.

Sometime it's profitable to inline small core scripts directly into page (like mobify, modernizr and yepnope).

**Comment by jansepar** [comment]
Hi @puzrin, sorry I haven't gotten back to you on this! I should hopefully have time to hack this together soon since it shouldn't be much work :)

**Comment by puzrin** [comment]
Thanks for good news. I've temporary rolled back mobifier use due side effects in FF. Plan to try again when situation with 2.0 stabilises.

Just one more note about intergation with other projects. Please, provide separate file with loader code (that loads mobifier). When project use assets manager (like sprockets/mincer), it's more practical to include loader with marcos, than manualy copy-paste from readme something minified.

**Comment by jansepar** [comment]
@puzrin what kind of side effects were you running into with FF? Just want to make sure there aren't issues I'm not aware of.

**Comment by puzrin** [comment]
Last time, when i used mobifier, where was several issues and i had to roll back due lack of time:
1. My project uses history API very intensively. So, "problem with back button" caused global fuckup in FF - nothing worked at all. I'm sure, that workaround is possible, but had no time to dig deeply.
2. lazyoad plugin for images had bug, that caused existing images reload on every scroll action
3. Mobifier was expected for "manual" integration with existing projects (no separate loader file, no possibility to include mobifier directly into page).

So, i decided to take a timeout - i've played with 2.0alfa14, and thinked that it's not right time to bother you with my problems :)

**Comment by jansepar** [comment]
@puzrin interesting that you had problems with the back button. That is a solved issue! If you have anymore details feel free to shoot them over :)

**Comment by puzrin** [comment]
Glad to see that FF issue is solved now. Was it fixed in firefox or you added hack to mobifier? Can't find commit in history.

**Comment by jansepar** [comment]
@puzrin the fix was resolved a very long time ago, at a time that was likely before you used it, so there very well could be an issue I have never encountered. If you had a minimal test case I would love to take a look.

**Comment by hallvors** [comment]
The project @puzrin is working on combined with Mobify.js might very well be the first attempt at using history.pushState() with document.write() - generated documents in Firefox. I'm not at all surprised if this would cause some issues because it probably is barely tested - if at all. That said, I would also be interested in seeing a minimal test case - if we have Firefox bugs here it would be good to be aware of them.

**Comment by jansepar** [comment]
@hallvors we became aware of the bug with mobify.js and history.pushState() in Firefox a couple weeks ago. We also became aware of an issue that caused Firefox to refresh the page when clicking on an anchor with hash links causes. We came up with a fix for that here, but its sub-optimal: 

http://www.mobify.com/mobifyjs/v2/docs/capturing/#patchanchorlinks

That page also mentions the issue with the history API (we only noticed it when we were attempting to create a patch for the anchor issue). I have it in my TODO list to submit a minimal test case to bugzilla - I should hopefully have that this week sometime.

We also have another bug open for the refresh issue we have in FF: https://bugzilla.mozilla.org/show_bug.cgi?id=556002 (although I just notice you've already commented on it!). It would be great to get more visibility for that issue.

**Comment by hallvors** [comment]
@jansepar great if you can report that bug and send over a minimal test :+1: 

I noticed the refresh problem while working on https://bugzilla.mozilla.org/show_bug.cgi?id=944508 and actually also noticed that you had a workaround coming up - https://bugzilla.mozilla.org/show_bug.cgi?id=881487#c10 

I will certainly try to _help_ getting bugs that affect you killed, but as I unfortunately don't fix them myself I can't give you any real ETA. Also if you have any input on the other wired.com problems it would help me a lot ;-)

**Comment by jansepar** [comment]
Ah, cool, so there is a bug report already for the anchor link issue! I've voted on it :)

The patch I submitted has only been added to mobify.js 2.0 and has not been backported to 1.1 (which is what Wired is using). The patch could be backported but it would be highly preferred for this to be fixed in FF itself, as our patch for it breaks anchor links showing up in the URL bar, and due to the history API being borked, we can't patch that.
